catalog/registry/catalog-registry-cswinputtransformer/src/main/java/ddf/catalog/registry/transformer/RegistryServiceConverter.java

```diff
@@ -0,0 +1,441 @@
+
+public class RegistryServiceConverter implements Converter {
+
+    private final Gml3ToWkt gml3ToWkt;
+
+    private static final RegistryServiceMetacardType RSMT = new RegistryServiceMetacardType();
+
+    private static final Map<String, String> CLASSIFICATIONS;
+
+    static {
+        CLASSIFICATIONS = new HashMap<>();
+        CLASSIFICATIONS.put("liveDate", RegistryServiceMetacardType.LIVE_DATE);
+        CLASSIFICATIONS.put("dataStartDate", RegistryServiceMetacardType.DATA_START_DATE);
+        CLASSIFICATIONS.put("dataEndDate", RegistryServiceMetacardType.DATA_END_DATE);
+        CLASSIFICATIONS.put("links", RegistryServiceMetacardType.LINKS);
+        CLASSIFICATIONS.put("region", RegistryServiceMetacardType.REGION);
+        CLASSIFICATIONS.put("dataRegion", RegistryServiceMetacardType.DATA_REGION);
+        CLASSIFICATIONS.put("inputDataSources", RegistryServiceMetacardType.DATA_SOURCES);
+        CLASSIFICATIONS.put("dataTypes", RegistryServiceMetacardType.DATA_TYPES);
+        CLASSIFICATIONS.put("securityLevel", RegistryServiceMetacardType.SECURITY_LEVEL);
+    }
+
+    public RegistryServiceConverter(Gml3ToWkt gml3ToWkt) {
+        this.gml3ToWkt = gml3ToWkt;
+    }
+
+    @Override
+    public void marshal(Object o, HierarchicalStreamWriter hierarchicalStreamWriter,
+            MarshallingContext marshallingContext) {
+    }
+
+    /**
+     * Read and format address info from a rim:Address node.
+     *
+     * @param reader
+     * @return Formatted address
+     */
+    private String readAddress(HierarchicalStreamReader reader) {
+        String street = reader.getAttribute("street");
+        String city = reader.getAttribute("city");
+        String stateOrProvince = reader.getAttribute("stateOrProvince");
+        String postalCode = reader.getAttribute("postalCode");
+        String country = reader.getAttribute("country");
+
+        return String
+                .format("%s, %s, %s %s, %s", street, city, stateOrProvince, postalCode, country);
+    }
+
+    /**
+     * Read and format phone info from a rim:TelephoneNumber node.
+     *
+     * @param reader
+     * @return Formatted phone number
+     */
+    private String readPhoneNumber(HierarchicalStreamReader reader) {
+        String areaCode = reader.getAttribute("areaCode");
+        String number = reader.getAttribute("number");
+        String extension = reader.getAttribute("extension");
+
+        if (StringUtils.isEmpty(extension)) {
+            return String.format("(%s) %s", areaCode, number);
+        } else {
+            return String.format("(%s) %s extension %s", areaCode, number, extension);
+        }
+    }
+
+    /**
+     * Read organization info from a rim:Organization node and attach to metacard.
+     *
+     * @param reader
+     * @param meta
+     */
+    private void readOrgInfo(HierarchicalStreamReader reader, RegistryMetacardImpl meta) {
+
+        reader.moveDown();
```

Comment:
I was thinking about other ways to ensure you always go up and down the same amount, here's what I came up with as a first draft:
```
private void safeMove(HierarchicalStreamReader reader, Integer amount, Runnable callback){
   for(int i = 0; i<amount; i++){
       reader.moveDown();
   }
   try {
       callback.run();
   } catch (Exception exception) {
       //something went wrong
   } finally {
       for(int i =0; i< amount; i++){
           reader.moveUp();
       }
   }
}

private void readOrgInfo(HierarchicalStreamReader reader, RegistryMetacardImpl meta){
   safeMove(reader, 3, () -> {
       while(reader.hasMoreChildren()){
           safeMove(reader, 1, () -> {
               switch (reader.getNodeName()) {
               case "rim:Name":
                   meta.setAttribute(RegistryObjectMetacardType.ORGANIZATION_NAME, readValue(reader));
                   break;
               case "rim:EmailAddress":
                   meta.setAttribute(RegistryObjectMetacardType.ORGANIZATION_EMAIL,
                           reader.getAttribute("address"));
                  break;
               case "rim:TelephoneNumber":
                   meta.setAttribute(RegistryObjectMetacardType.ORGANIZATION_PHONE_NUMBER,
                           readPhoneNumber(reader));
                   break;
               case "rim:Address":
                   meta.setAttribute(RegistryObjectMetacardType.ORGANIZATION_ADDRESS,
                           readAddress(reader));
                   break;
               }
           });
       }
   });
}
```